src/workerd/jsg/promise.h

```diff
@@ -374,6 +374,21 @@ class Promise {
     }
   }
 
+  // Ths is for testing/diagnostics purposes only.
```

Comment:
Perhaps it should be private and then make the testing class a friend?

Reply:
It looks like `getState()` is called from within `KJ_TEST`, which need to be defined in anonymous namespaces to avoid linker errors. So, any escape hatch might end up being more or less equivalent to this.